examples/Container/Frontend/Program.cs

```diff
@@ -16,32 +16,71 @@
 
 #endregion
 
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
+using Frontend.Balancer;
+using Grpc.Core;
+using Grpc.Net.Client;
+using Grpc.Net.Client.Balancer;
 
-namespace Frontend
+var builder = WebApplication.CreateBuilder(args);
+builder.Logging.SetMinimumLevel(LogLevel.Trace);
+builder.Logging.AddSimpleConsole(c =>
 {
-    public class Program
+    c.TimestampFormat = "[HH:mm:ss.ff]";
+});
+builder.Services.AddRazorPages();
+builder.Services.AddServerSideBlazor();
+
+builder.Services.AddSingleton(services =>
+{
+    var backendUrl = builder.Configuration["BackendUrl"]!;
+
+    var channel = GrpcChannel.ForAddress(backendUrl, new GrpcChannelOptions
+    {
+        Credentials = ChannelCredentials.Insecure,
+        ServiceProvider = services
+    });
+
+    return channel;
+});
+
+SetupReportingServices(builder.Services);
```

Comment:
Super nit: I'd almost prefer (a) having another file to hold the extension method or (b) doing the registrations inline to this since it's not a common pattern. I know it's just an example but people tend to copy-paste from these a lot.

Reply:
Done